src/dbnode/persist/schema/types.go

```diff
@@ -49,39 +49,40 @@ type IndexInfo struct {
 	MinorVersion int64
 }
 
-// IndexSummariesInfo stores metadata about the summaries
+// IndexSummariesInfo stores metadata about the summaries.
 type IndexSummariesInfo struct {
 	Summaries int64
 }
 
-// IndexBloomFilterInfo stores metadata about the bloom filter
+// IndexBloomFilterInfo stores metadata about the bloom filter.
 type IndexBloomFilterInfo struct {
 	NumElementsM int64
 	NumHashesK   int64
 }
 
-// IndexEntry stores entry-level data indexing
+// IndexEntry stores entry-level data indexing.
 //
 // When serialized to disk, the encoder will automatically add the IndexEntryChecksum, a checksum to validate
 // the index entry itself, to the end of the entry. That field is not exposed on this struct as this is handled
 // transparently by the encoder and decoder. Appending of checksum starts in V3.
 type IndexEntry struct {
-	Index        int64
-	ID           []byte
-	Size         int64
-	Offset       int64
-	DataChecksum int64
-	EncodedTags  []byte
+	Index         int64
+	ID            []byte
+	Size          int64
+	Offset        int64
+	DataChecksum  int64
+	EncodedTags   []byte
+	IndexChecksum uint32
```

Comment:
Might be worth adding a comment here explaining what the `IndexChecksum` represents. Could see folks being confused trying to understand the difference between this value and the implicitly added `IndexEntryChecksum`

Reply:
Ah yeah, so this is the real `IndexChecksum` that's calculated by hashing all the entry fields; before we were computing these but discarding the results. Was attempting to use `IndexChecksum` as the hash function previously but that had a few issues; right now for wide queries you need to use the Hash. Will definitely add some comments, but this is technically unused at the moment, can revert the change and have it be discarded after the checksum comparison on read

Reply:
Also is it worth using an `int64` here to be consistent with other places (and since msgpack is going to return you an `int64` anyway when you call `decodeVarint`)?

Reply:
Yeah good call, will update it to `int64` unless we just want to drop the field again